src/mumble/UserModel.cpp

```diff
@@ -1958,18 +1958,32 @@ QString UserModel::createDisplayString(const ClientUser &user, bool isChannelLis
 	if (!user.qsFriendName.isEmpty() && user.qsName.compare(user.qsFriendName, Qt::CaseInsensitive) != 0) {
 		friendTag = QString::fromLatin1("(%2)").arg(user.qsFriendName);
 	}
+	
+	// Create a nickname-tag
+	QString nickname = user.getLocalNickname();
+	QString nicknameTag;
+	if (!nickname.isEmpty() && user.qsName.compare(nickname, Qt::CaseInsensitive) != 0) {
+		nicknameTag = QString::fromLatin1("[%2]").arg(nickname);
+	}
 
 	// Create a tag that indicates the volume adjustments
 	QString volumeTag;
 	if (std::abs(localVolumeDecibel) > 0 && g.s.bShowVolumeAdjustments) {
 		volumeTag = QString::asprintf("|%+d|", localVolumeDecibel);
 	}
 
-	QString displayString = user.qsName;
+	QString displayString;
+	if (!g.s.bShowNicknamesOnly || nicknameTag.isEmpty()) {
+		displayString = user.qsName;
+	}
 
 	if (!friendTag.isEmpty()) {
 		displayString += " " + friendTag;
 	}
+	
+	if (!nicknameTag.isEmpty()) {
+		displayString += " " + nicknameTag;
+	}
```

Comment:
I think in case `bShowNicknamesOnly` it would make sense to not show the nickname as a tag, but rather as the normal name instead (aka: without the brackets around the name)